src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java

```diff
@@ -69,48 +69,55 @@ private static String getVoltageLevelFieldValue(FieldType field, VoltageLevel vo
                 VOLTAGE_LEVEL_ID_2 -> voltageLevel.getId();
             case LOW_VOLTAGE_LIMIT -> String.valueOf(voltageLevel.getLowVoltageLimit());
             case HIGH_VOLTAGE_LIMIT -> String.valueOf(voltageLevel.getHighVoltageLimit());
+            case FREE_PROPERTIES -> voltageLevel.getProperty(propertyName);
+            case SUBSTATION_PROPERTIES -> voltageLevel.getNullableSubstation().getProperty(propertyName);
             default -> throw new PowsyblException(FIELD_AND_TYPE_NOT_IMPLEMENTED + " [" + field + "," + voltageLevel.getType() + "]");
         };
     }
 
-    private static String getLineFieldValue(FieldType field, Line line) {
+    private static String getLineFieldValue(FieldType field, String propertyName, Line line) {
         return switch (field) {
             case CONNECTED_1 -> getTerminalFieldValue(field, line.getTerminal(TwoSides.ONE));
             case CONNECTED_2 -> getTerminalFieldValue(field, line.getTerminal(TwoSides.TWO));
             case COUNTRY_1,
                 VOLTAGE_LEVEL_ID_1,
-                NOMINAL_VOLTAGE_1 -> getVoltageLevelFieldValue(field, line.getTerminal(TwoSides.ONE).getVoltageLevel());
+                NOMINAL_VOLTAGE_1 -> getVoltageLevelFieldValue(field, null, line.getTerminal(TwoSides.ONE).getVoltageLevel());
             case COUNTRY_2,
                 VOLTAGE_LEVEL_ID_2,
-                NOMINAL_VOLTAGE_2 -> getVoltageLevelFieldValue(field, line.getTerminal(TwoSides.TWO).getVoltageLevel());
+                NOMINAL_VOLTAGE_2 -> getVoltageLevelFieldValue(field, null, line.getTerminal(TwoSides.TWO).getVoltageLevel());
             case SERIE_RESISTANCE -> String.valueOf(line.getR());
             case SERIE_REACTANCE -> String.valueOf(line.getX());
             case SHUNT_CONDUCTANCE_1 -> String.valueOf(line.getG1());
             case SHUNT_CONDUCTANCE_2 -> String.valueOf(line.getG2());
             case SHUNT_SUSCEPTANCE_1 -> String.valueOf(line.getB1());
             case SHUNT_SUSCEPTANCE_2 -> String.valueOf(line.getB2());
+            case FREE_PROPERTIES -> line.getProperty(propertyName);
```

Comment:
not needed as we add a case in getFieldValue

Reply:
done